pkg/bastion/shell.go

```diff
@@ -1581,6 +1581,59 @@ GLOBAL OPTIONS:
 			Usage: "Manages users",
 			Subcommands: []cli.Command{
 				{
+					Name:        "create",
+					ArgsUsage:   "<email>",
+					Usage:       "Creates a new user",
+					Description: "$> user create bob@example.com\n   $> user create --name=Robert bob@example.com",
+					Flags: []cli.Flag{
+						cli.StringFlag{Name: "name", Usage: "Assigns a name to the user"},
+						cli.StringFlag{Name: "comment", Usage: "Adds a comment"},
+						cli.StringSliceFlag{Name: "group, g", Usage: "Names or IDs of `USERGROUPS` (default: \"default\")"},
+					},
+					Action: func(c *cli.Context) error {
+						if c.NArg() != 1 {
+							return cli.ShowSubcommandHelp(c)
+						}
+
+						if err := myself.CheckRoles([]string{"admin"}); err != nil {
+							return err
+						}
+
+						// FIXME: validate email
+
+						email := c.Args().First()
```

Comment:
@jwessel We can validate email with regexp 
```go
var emailRegex = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
```
So after that we can make helper func and try to match with the helper method from emailRegex on input string(email)

Reply:
I move your idea as a more global issue, since we need to validate email in various places which makes your idea of using a wrapper very good :+1:

Reply:
-> https://github.com/moul/sshportal/issues/259#issuecomment-808302445